activerecord/lib/active_record/attribute_methods/primary_key.rb

```diff
@@ -21,6 +21,12 @@ def id
         @primary_key.map { |pk| _read_attribute(pk) }
       end
 
+      def primary_key_values_present? # :nodoc:
+        return id.size > 0 && id.all? if self.class.composite_primary_key?
```

Comment:
I don't think size is correct here. When we have an ID like `1` the size will be `8`. An array of `[1,2]` will return `2`. sqlite3 tests pass without the `id.size > 0` check, if there's a reason we need it, can you add a test for it?